packages/server/src/Connection.ts

```diff
@@ -216,21 +220,40 @@ export class Connection {
    * @public
    */
   public handleMessage(data: Uint8Array): void {
+    if (this.beforeHandleMessageQueue.length > 0 || this.beforeHandleMessageProcessing) {
+      this.beforeHandleMessageQueue.push(data)
+      return
+    }
+
+    this.processBeforeHandleMessage(data)
+  }
+
+  /** Process an incoming message */
+  private processBeforeHandleMessage(data: Uint8Array): void {
```

Comment:
The name of the method is out of sync with the comment. I like the comment more, as what this methos does is process an incoming message after applying the beforeHandleMessage hooks.